databook/physiology/ophys/visual-behavior/VBO-ExperimentData.md

```diff
@@ -433,38 +461,81 @@ dataframe columns:
+
+movie_frame_index
+: (int) index corresponding to the frame in the natural movie that was displayed at the end of some sessions; value will be -99 for all stimulus blocks other than `natural_movie_one`
+
+duration
+: (float) the duration, in seconds, of each stimulus presentation or spontaneous activity period; equivalent to `stop_time` - `start_time`
+
 start_time
 : (float) image presentation start time in seconds
 
 end_time
 : (float) image presentation end time in seconds
 
-image_name
-: *str* Indicates which natural scene image stimulus was shown for this stimulus presentation.
+start_frame
+: (int) image presentation start frame
 
-image_index
-: (int) image index (0-7) for a given session, corresponding to each image name
+end_frame
+: (float) image presentation end frame
 
 is_change
-: *bool* Indicates whether the image identity changed for this stimulus presentation
+: (bool) whether the stimulus identity changed for this stimulus presentation
+
+is_image_novel
+: (bool) whether the stimulus presentation is an image from the novel (untrained) image set (does not indicate the very first presentation of a given image)
 
 omitted
 : (bool) True if no image was shown for this stimulus presentation
 
+movie_repeat
+: (int) index indicating the order of repeatitions of the 30 second movie clip in the natural_movie_one stimulus block (0-9); value will be -99 for other stimulus blocks
+ 
+flashes_since_change
+: *float* indicates how many flashes of the same image have occurred since the last stimulus change; values will be NaN for stimulus blocks other than `change_detection_behavior`
+
 trials_id
-: (int) Id to match to the table Index of the trials table.
+: (int) Id to match to the table Index of the `trials` table
 
-flashes_since_change
-: *float* Indicates how many flashes of the same image have occurred since the last stimulus change. NaN for blocks 1-4.
+active
+: (bool) whether the stimulus presentation is part of an active behavior block (i.e mouse performing change detection task, not passive replay)
 
-start_frame
-: (int) image presentation start frame
+stimulus_name
+: (str) name of the stimulus set used for the current stimulus block 
 
-end_frame
-: (float) image presentation end frame
+is_sham_change
+: (bool) whether the current stimulus presentation could have been a change based on the change time distribution; can be considered as "no-go" trials and used to quantify false alarms 
 
-duration
-: (float) duration of an image presentation (flash) in seconds (stop_time - start_time). NaN if omitted
+
+Let's explore the `stimulus_presentations` for the `natural_movie_one` stimulus block. Select the `natural_movie_one` block and inspect the unique values of `image_name`, `movie_repeat`, and `movie_frame_index`.
+
+```{code-cell} ipython3
+movie_stimulus_table = stimulus_table[stimulus_table.stimulus_block_name=='natural_movie_one']
+
+print('Values of `image_name` are:', movie_stimulus_table.unique())
```

Comment:
This code cell fails.

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_1993/181777896.py in ?()
      1 movie_stimulus_table = stimulus_table[stimulus_table.stimulus_block_name=='natural_movie_one']
      2 
----> 3 print('Values of `image_name` are:', movie_stimulus_table.unique())
      4 print('Values of `movie_repeat` are:', movie_repeat.unique())
      5 print('Values of `movie_frame_index` are:', movie_frame_index.unique())

/opt/envs/allensdk/lib/python3.10/site-packages/pandas/core/generic.py in ?(self, name)
   5898             and name not in self._accessors
   5899             and self._info_axis._can_hold_identifiers_and_holds_name(name)
   5900         ):
   5901             return self[name]
-> 5902         return object.__getattribute__(self, name)

AttributeError: 'DataFrame' object has no attribute 'unique'
```

Reply:
sorry about that, fixed in recent commit